homeassistant/components/modbus/validators.py

```diff
@@ -421,6 +447,10 @@ def validate_entity(
         loc_addr.add(f"{hub_name}{entity[CONF_HVAC_MODE_REGISTER][CONF_ADDRESS]}_{inx}")
     if CONF_FAN_MODE_REGISTER in entity:
         loc_addr.add(f"{hub_name}{entity[CONF_FAN_MODE_REGISTER][CONF_ADDRESS]}_{inx}")
+    if CONF_SWING_MODE_REGISTER in entity:
+        loc_addr.add(
+            f"{hub_name}{entity[CONF_SWING_MODE_REGISTER][CONF_ADDRESS]}_{inx}"
```

Comment:
This need to be different for a list.

Reply:
Dear @janiversen , you're right. But the same should be for ` loc_addr.add(f"{hub_name}{entity[CONF_FAN_MODE_REGISTER][CONF_ADDRESS]}_{inx}")` as I tried to tell you respectfully at #108906 (https://github.com/home-assistant/core/pull/108906#discussion_r1467787468)
Similar (but not equal, because for hvac modes you could have duplicates) case in `loc_addr.add(f"{hub_name}{entity[CONF_TARGET_TEMP]}_{inx}")` . 

Moreover, IMHO it seems that the check_config and validate_entity work bad. In example, from line 537 of the test_init file:
`[
            {
                CONF_NAME: TEST_MODBUS_NAME,
                CONF_TYPE: TCP,
                CONF_HOST: TEST_MODBUS_HOST,
                CONF_PORT: TEST_PORT_TCP,
                CONF_TIMEOUT: 3,
                CONF_CLIMATES: [
                    {
                        CONF_NAME: TEST_ENTITY_NAME,
                        CONF_ADDRESS: 118,
                        CONF_SLAVE: 0,
                        CONF_HVAC_MODE_REGISTER: {
                            CONF_ADDRESS: 119,
                            CONF_HVAC_MODE_VALUES: {
                                CONF_HVAC_MODE_COOL: 0,
                                CONF_HVAC_MODE_HEAT: 1,
                            },
                        },
                    },
                    {
                        CONF_NAME: TEST_ENTITY_NAME + " 2",
                        CONF_ADDRESS: 117,
                        CONF_SLAVE: 0,
                        CONF_HVAC_MODE_REGISTER: {
                            CONF_ADDRESS: 118,
                            CONF_HVAC_MODE_VALUES: {
                                CONF_HVAC_MODE_COOL: 0,
                                CONF_HVAC_MODE_HEAT: 1,
                            },
                        },
                    },
                ],
            }
        ],` 

Here the second entity is discarded because the hvac mode register of the Entity 2 overlaps with the conf_address of the entity 1 and it should be not an error, because they are 2 distinct entities.. 
 
It is reasonable to think that the centralized_validation should be adjusted, don't you think? 

Please, advice me in case I'm missing something..

Reply:
The central validation do take input_type in account, but NOT entity, and that is correct.

e.g. A holding register address must be unique across entities.

Reply:
My bad, excuse me. The discriminanting parameter is CONF_SLAVE.
I'll add an if for the list. In another PR I'll address the same wrong behaviour for `{entity[CONF_FAN_MODE_REGISTER][CONF_ADDRESS]}_{inx}")`

Reply:
I have no idea what you say.....you should not the address as a list, that will not compare correctly against other addresses

Reply:
[1] and 1 should both be added as 1 in order to detect if 1 is used by another entity,

Reply:
Yes, undestood: https://github.com/home-assistant/core/pull/113710/commits/58aac8390c696d0b3ae4beed1f97e57103d19aa1